src/awsHttpHandler/httpHandler.ts

```diff
@@ -0,0 +1,116 @@
+import { hckFetch } from '../index';
+import { requestTimeout } from './requestTimeout';
+import { isEmpty } from './isEmpty';
+
+type LogHandler = (...content: unknown[]) => string;
+
+export type Logger = {
+  log?: LogHandler;
+  info?: LogHandler;
+};
+
+export type HttpHandlerConstructor = {
+  requestTimeout?: number;
+  logger?: Logger;
+};
+
+type RequestOptions = {
+  headers: Headers;
+  body?: any;
+  method: string;
+  signal?: AbortSignal;
+};
+
+export class HttpHandler {
```

Comment:
You should implement [the interface from aws-sdk](https://github.com/aws/aws-sdk-js-v3/blob/5d9bc5f32980f43a9b581b1f5101cdc7f2258173/packages/fetch-http-handler/src/fetch-http-handler.ts#L20) and reuse the appropriate types.

```suggestion
export class HckFetchHttpHandler implements HttpHandler {
```

Note that the interface belongs to [@aws-sdk/protocol-http](https://www.npmjs.com/package/@aws-sdk/protocol-http) that has been deprecated in favour of [@smithy/protocol-http](https://www.npmjs.com/package/@smithy/protocol-http).